src/analyzer/protocol/ldap/ldap.spicy

```diff
@@ -670,6 +670,27 @@ public function bytes_sid_to_hex_repr(bts: bytes): string {
     return ret;
 }
 
+public function escape_filter_string(bts: bytes): string {
+    local ret = "";
+    local cnt = 0;
+
+    while (cnt < |bts|) {
```

Comment:
We could use a `for` loop here to avoid manual counter management, and having to get the value out of `bts`.

```ruby
    for (bt in bts) { ... }
```